### PR TITLE
CAMEL-11754 - add option to extract file name if ftp client directory

### PR DESCRIPTION
**Pull request by onders86**
parser ends up with absolute paths

### CODE REVIEW

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpUtils.java [hunk 1]
```diff
@@ -35,6 +35,19 @@
...
+        // default is unix so try with '/'
+        // otherwise force File.separator
+        if (path.endsWith("/") || path.endsWith(File.separator)) {
+            path = path.substring(0, path.length() - 1);
+        }
+        int startIndex = path.lastIndexOf(File.separator);
+        if (startIndex == -1) {
+            startIndex = path.lastIndexOf("/");
+        }
+        return path.substring(startIndex + 1, path.length());
```
Comment (davsclaus) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpUtils.java]: If there is no slash then start index is -1 and then I think we should make it more clear that we dont substring it then, eg

if (startIndex != -1) {
   ..
} else {
   // no directory then return file as-is
   return path;
}

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpConfiguration.java [hunk 2]
```diff
@@ -35,6 +35,8 @@
     private String activePortRange;
     @UriParam(label = "producer,advanced")
     private String chmod;
+    @UriParam(label = "consumer,advanced")
+    private boolean isDirectoryParseResultAbsolute;
```
Comment (davsclaus) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpConfiguration.java]: dont use isXXX as the name. Naming in IT is hard, and to come up with a good name.
Maybe the name should be something like `handleDirectoryParserAbsoluteResult` ?

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpConfiguration.java [hunk 3]
```diff
@@ -83,4 +85,16 @@ public void setChmod(String chmod) {
...
         return chmod;
     }
+
+    public boolean isDirectoryParseResultAbsolute() {
+        return isDirectoryParseResultAbsolute;
+    }
+
+    /**
+     * Allows you to set how the consumer will handle subfolders and files
+     * in the path if the directory parser results in with absolute paths
```
Comment (davsclaus) [inline components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpConfiguration.java]: Maybe add a comment: Some FTP servers may return file names with absolute paths, and if so then the FTP component needs to handle this by converting the returned path into a relative path.

File: components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpUtils.java (hunk 1 above)
Comment (davsclaus) [reply components/camel-ftp/src/main/java/org/apache/camel/component/file/remote/FtpUtils.java]: Also there is a stripPath method from FileUtil you can use for the last part: org.apache.camel.util.FileUtil#stripPath

**Comment by dheitzerLLY** [comment]
Thanks for the help guys.  It looks good.

**Pull request merged**